File: src/frontend/quiz/src/types/UserTypes.ts
export interface User {
  info: UserInfo
  credentials: Credentials
}

export interface UserInfo {
  username: string
  name: string
  role: Role
  token: string
}

export interface Credentials {
  username: string
  password: string
}

export enum Role {
  NONE = 'none',
  ADMIN = 'admin',
  TEACHER = 'teacher',
  STUDENT = 'student'
}

File: src/gateway_api_service/UserTypes.py
from enum import Enum

from pydantic import BaseModel


class Credentials(BaseModel):
    username: str
    password: str

class Role(str, Enum):
    ADMIN = 'admin',
    TEACHER = 'teacher'
    STUDENT = 'student'

class UserInfo(BaseModel):
    username: str
    name: str
    role: Role
    token: str

class User(BaseModel):
    info: UserInfo
    credentials: Credentials



